internal/service/advancedcluster/data_source_advanced_cluster.go

```diff
@@ -264,7 +264,7 @@ func dataSourceRead(ctx context.Context, d *schema.ResourceData, meta any) diag.
 	}
 
 	if !useReplicationSpecPerShard {
-		clusterDescOld, resp, err := connV2.ClustersApi.GetCluster(ctx, projectID, clusterName).Execute()
+		clusterDescOld, resp, err := connV220231115.ClustersApi.GetCluster(ctx, projectID, clusterName).Execute()
```

Comment:
nit for future refactoring: maybe we can use some [strategy pattern](https://en.wikipedia.org/wiki/Strategy_pattern) here? it could make the code cleaner and less relying on "ifs" 

if you agree, always good to track with TODO and CLOUDP

Reply:
Yes this could be an option for removing nested logic, applies for the read operation of the resource as well. On the other hand I do see that the logic within the if/else is quite simple so could be over complicating the solution. Will create a ticket so we can track and revisit this.